providers/ModalProvider.js

```diff
@@ -49,6 +49,10 @@ function modalReducer(state, action) {
         ...state,
         activeModal: {
           ...state.activeModal,
+          props: {
+            ...state.activeModal.props,
+            step: action.payload.step,
+          },
```

Comment:
Curious what you think about this, @drewbarontini.
This fixes a small bug I found with the `ModalProvider` mechanics.

If you open a modal to a specific step like this:
```javascript
modalDispatch(showModal('GroupFilter', { step: 1 }));
```
And subsequently try to navigate to the next step like this:
```javascript
modalDispatch(showStep(2));
```

The initial `{ step: 1 }` value gets "stuck" and the modal doesn't render the new screen.
The action gets reduced, but the `Modal` component gets passed `props.step` instead of the `state.step`.

This fix doesn't feel stupendous, but I think it's safe and reasonable.
There may be a smarter/better one.

Reply:
I think this makes sense and can't think of something better at the moment 👍